multiqc/modules/pangolin/pangolin.py

```diff
@@ -0,0 +1,164 @@
+#! /usr/bin/env python
+
+""" MultiQC module to parse output from Pangolin """
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+import csv
+
+from multiqc import config
+from multiqc.plots import beeswarm, table
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    """Pangolin module"""
+
+    def __init__(self):
+
+        # Initialise the parent module
+        super().__init__(
+            name="Pangolin",
+            anchor="pangolin",
+            href="https://github.com/cov-lineages/pangolin",
+            info="uses variant calls to assign SARS-CoV-2 genome sequences to global lineages.",
+        )
+
+        # Find and parse the sample files
+        self.pangolin_data = dict()
+        for f in self.find_log_files("pangolin", filehandles=True):
+            self.parse_pangolin_log(f)
+
+        # Filter out parsed samples based on sample name
+        self.pangolin_data = self.ignore_samples(self.pangolin_data)
+
+        # Stop if we didn't find anything
+        if len(self.pangolin_data) == 0:
+            raise UserWarning
+
+        self.pangolin_general_stats_table()
+
+        self.add_section(
+            name="Run table",
+            anchor="pangolin-run-table",
+            description="Statistics gathered from the input pangolin files",
+            plot=self.pangolin_table(),
+        )
+
+    def parse_pangolin_log(self, fh):
+        contents = csv.DictReader(fh["f"])
```

Comment:
This is in a core Python library? I don't think I've ever used this before 🤯 

Learn something new every day! 😆